.github/workflows/REVIEW_GUIDELINES.md

```diff
@@ -0,0 +1,712 @@
+
+### Class defaults
+
+By default, we don't use `final`. For our open source stuff, we assume that all our users know they are responsible for writing tests for any overwritten behaviour.
+
+### Void return types
+
+If a method return nothing, it should be indicated with `void`. 
+This makes it more clear to the users of your code what your intention was when writing it.
+
+## Typed properties
+
+You should type a property whenever possible. Don't use a docblock.
+
+```php
+// good
+class Foo
+{
+    public string $bar;
+}
+
+// bad
+class Foo
+{
+    /** @var string */
+    public $bar;
+}
+```
+
+## Docblocks
+
+Don't use docblocks for methods that can be fully type hinted (unless you need a description).
+
+Only add a description when it provides more context than the method signature itself. Use full sentences for descriptions, including a period at the end.
+
+```php
+// Good
+class Url
+{
+    public static function fromString(string $url): Url
+    {
+        // ...
+    }
+}
+
+// Bad: The description is redundant, and the method is fully type-hinted.
+class Url
+{
+    /**
+     * Create a url from a string.
+     *
+     * @param string $url
+     *
+     * @return \Spatie\Url\Url
+     */
+    public static function fromString(string $url): Url
+    {
+        // ...
+    }
+}
+```
+
+Always use fully qualified class names in docblocks.
+
+```php
+// Good
+
+/**
+ * @param string $url
+ *
+ * @return \Spatie\Url\Url
+ */
+
+// Bad
+
+/**
+ * @param string $foo
+ *
+ * @return Url
+ */
+```
+
+Docblocks for class variables are required, as there's currently no other way to typehint these.
+
+```php
+// Good
+
+class Foo
+{
+    /** @var \Spatie\Url\Url */
+    private $url;
+
+    /** @var string */
+    private $name;
+}
+
+// Bad
+
+class Foo
+{
+    private $url;
+    private $name;
+}
+```
+
+When possible, docblocks should be written on one line.
+
+```php
+// Good
+
+/** @var string */
+/** @test */
+
+// Bad
+
+/**
+ * @test
+ */
+```
+
+If a variable has multiple types, the most common occurring type should be first.
+
+```php
+// Good
+
+/** @var \Spatie\Goo\Bar|null */
+
+// Bad
+
+/** @var null|\Spatie\Goo\Bar */
+```
+
+## Strings
+
+When possible prefer string interpolation above `sprintf` and the `.` operator.
+
+```php
+// Good
+$greeting = "Hi, I am {$name}.";
+```
+
+```php
+// Bad
+$greeting = 'Hi, I am ' . $name . '.';
+```
+
+
+## Ternary operators
+
+Every portion of a ternary expression should be on its own line unless it's a really short expression.
+
+```php
+// Good
+$result = $object instanceof Model
+    ? $object->name
+    : 'A default value';
+
+$name = $isFoo ? 'foo' : 'bar';
+
+// Bad
+$result = $object instanceof Model ?
+    $object->name :
+   'A default value';
+```
+
+## If statements
+
+### Bracket position
+
+Always use curly brackets.
+
+```php
+// Good
+if ($condition) {
+   ...
+}
+
+// Good: One line is acceptable if the next statement is not complex and only one line
```

Comment:
Avoid using single line `if` statements without braces. It can lead to confusion and bugs if more lines are added in the future.